Neos.ContentRepository/Classes/Domain/Model/Node.php

```diff
@@ -1833,11 +1905,268 @@ public function isAutoCreated()
+     */
+    public function findParentNode(): TraversableNodeInterface
+    {
+        /** @var TraversableNodeInterface $parentNode It's safe to return the old NodeInterface as TraversableNodeInterface; as the base implementation "Node" (this class) implements both interfaces at the same time. */
+        $parentNode = $this->getParent();
+        if ($parentNode === null) {
+            throw new NodeException('Parent node not found', 1542983610);
+        }
+        return $parentNode;
+    }
+
+    public function findNodePath(): NodePath
+    {
+        return NodePath::fromString($this->getPath());
+    }
+
+    /**
+     * @param NodeName $nodeName
+     * @return TraversableNodeInterface
+     * @throws NodeException
+     */
+    public function findNamedChildNode(NodeName $nodeName): TraversableNodeInterface
+    {
+        /** @var TraversableNodeInterface $childNode It's safe to return the old NodeInterface as TraversableNodeInterface; as the base implementation "Node" (this class) implements both interfaces at the same time. */
+        $childNode = $this->getNode((string)$nodeName);
+        if ($childNode === null) {
+            throw new NodeException(sprintf('Child node named "%s" not found', $nodeName), 1543406006);
+        }
+        return $childNode;
+    }
+
+    /**
+     * Returns all direct child nodes of this node.
+     * If a node type is specified, only nodes of that type are returned.
+     *
+     * @param NodeTypeConstraints $nodeTypeConstraints If specified, only nodes with that node type are considered
+     * @param int $limit An optional limit for the number of nodes to find. Added or removed nodes can still change the number nodes!
+     * @param int $offset An optional offset for the query
+     * @return TraversableNodes
+     * @api
+     */
+    public function findChildNodes(NodeTypeConstraints $nodeTypeConstraints = null, int $limit = null, int $offset = null): TraversableNodes
+    {
+        /** @noinspection PhpDeprecationInspection */
+        $filter = $nodeTypeConstraints !== null ? $nodeTypeConstraints->asLegacyNodeTypeFilterString() : null;
+        // It's safe to return the old NodeInterface as TraversableNodeInterface; as the base implementation "Node" (this class) implements both interfaces at the same time.
+        return TraversableNodes::fromArray($this->getChildNodes($filter, $limit, $offset));
+    }
+
+    /**
+     * Returns the number of direct child nodes of this node from its subgraph.
+     *
+     * @param NodeTypeConstraints|null $nodeTypeConstraints
+     * @return int
+     */
+    public function countChildNodes(NodeTypeConstraints $nodeTypeConstraints = null): int
+    {
+        return count($this->getChildNodes($nodeTypeConstraints));
+    }
+
+    /**
+     * Retrieves and returns all nodes referenced by this node from its subgraph.
+     * If node type constraints are specified, only nodes of that type are returned.
```

Comment:
I see nothing about constraints in the code

Reply:
Good point, will fix :)

Reply:
totally right, will fix with next commit!